tests/unit/plugins/inventory/test_flightctl.py

```
@@ -521,5 +528,309 @@ def test_validate_device_partial_expression_with_missing_field(self):
+        config = MagicMock()
+        config.access_token = None
+        config.username = "admin"
+        config.password = "secret"
+        headers = _build_auth_headers(config)
+        self.assertIsNone(headers)
+
+    def test_no_basic_auth_ever_produced(self):
+        config = MagicMock()
+        config.access_token = None
+        config.username = "admin"
+        config.password = "secret"
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to: "password"

(S105)